docs/ideas/async-asset-loading-overview.md

```
@@ -0,0 +1,269 @@
+LoadingHandle<MeshAsset>   StreamMesh  (std::string_view path, ELoadPriority = ELoadPriority::Normal);
+
+// This is critical, will be used by engine->LoadSceneAsync.
+LoadingHandle<SceneAsset>  StreamScene (std::string_view path, ELoadPriority = ELoadPriority::Normal);
+
+LoadingHandle<AudioClip>   StreamAudioClip(std::string_view path, ELoadPriority = ELoadPriority::Normal);
+```
+
+For the streaming API, some engine-owned systems will be able to subscribe to the loading events, for instance, the `AudioMixerSystem` will subscribe to `AudioClip` loading events, so it can start playing the music as soon as the first chunk is loaded, and it can stop the loading if the user stops the music before it's fully loaded.
+
+> **What if I `StreamX` an already-loaded asset?** Return a pre-resolved `LoadingHandle<T>`. `IsReady()` is true immediately, `Take()` returns the cached `Ref<T>`, `Subscribe()` is a no-op (no chunks will fire, the asset is already in memory). Subscribers should register both a chunk callback and an on-complete callback so they handle both the streaming path (chunks fire, then complete) and the already-loaded path (complete fires once, no chunks). Doesn't re-stream, doesn't error, doesn't redirect to `LoadX`.
+
```

markdownlint-cli2 (0.22.1)

[warning] Blank line inside blockquote

(MD028, no-blanks-blockquote)